src/features/views/components/ViewColumnDialog/ColumnGallery.tsx

```diff
@@ -12,104 +21,242 @@ import choices, { ColumnChoice } from './choices';
-            <ListItem
-              key={index}
-              onClick={() => {
-                if (choiceContainerRef.current) {
-                  const element = choiceContainerRef.current.querySelector(
-                    `#category-${index}`
-                  );
-                  element?.scrollIntoView({ behavior: 'smooth' });
-                }
-              }}
-              sx={{ cursor: 'pointer', paddingY: 2 }}
-            >
-              <Typography>
+    <Box display="flex" flexDirection="column" height="100%">
+      <Box display="flex" justifyContent="space-between" padding={2}>
+        <Typography variant="h4">
+          <Msg id="misc.views.columnDialog.gallery.columns" />
+        </Typography>
+        <Box alignItems="center" display="flex">
+          <TextField
+            InputProps={{
+              endAdornment: <Search color="secondary" />,
+            }}
+            onChange={(evt) => {
+              setSearchString(evt.target.value);
+              if (evt.target.value === '') {
+                setIsSearching(false);
+                return;
+              }
+              setIsSearching(true);
+              search();
+            }}
+            placeholder={intl.formatMessage({
+              id: 'misc.views.columnDialog.gallery.searchPlaceholder',
+            })}
+            sx={{ paddingRight: 2 }}
+            value={searchString}
+            variant="outlined"
+          />
+          <Close
+            color="secondary"
+            fontSize="medium"
+            onClick={() => onClose()}
+            sx={{ cursor: 'pointer' }}
+          />
+        </Box>
+      </Box>
+      <Box display="flex" height="100%">
+        <Box
+          alignItems="center"
+          display={isMobile ? 'none' : 'flex'}
+          flexDirection="column"
+          justifyContent="space-between"
+          sx={{
+            overflowY: 'scroll',
+          }}
+          width="20%"
+        >
+          <List>
+            {categories.map((category, index) => (
+              <ListItem
+                key={index}
+                onClick={() => {
+                  setSearchString('');
+                  setIsSearching(false);
+                  if (choiceContainerRef.current) {
+                    const element = choiceContainerRef.current.querySelector(
+                      `#category-${index}`
+                    );
+                    element?.scrollIntoView({ behavior: 'smooth' });
+                  }
+                }}
+                sx={{ cursor: 'pointer', paddingY: 2 }}
+              >
+                <Typography>
+                  <Msg
+                    id={`misc.views.columnDialog.categories.${category.key}.title`}
+                  />
+                </Typography>
+              </ListItem>
+            ))}
+          </List>
+        </Box>
+        <Box ref={choiceContainerRef} flexGrow={1} sx={{ overflowY: 'scroll' }}>
+          {isSearching ? (
```

Comment:
This double (inner/outer) ternary that stretches across 100 or so lines makes it very difficult for me to understand the logic. The double ternary is ok IMO, if the contents were refactored out into separate components.

Reply:
yes it is a horror to the eyes 😅 will refactor into two separate components!